src/share.tsx

```diff
@@ -17,10 +19,65 @@ function Header(props: PropsWithChildren): JSX.Element {
   )
 }
 
-export function ShareSpace (): JSX.Element {
+function isDID(value: string): boolean {
+  try {
+    DID.parse(value.trim())
+    return true
+  } catch (err) {
+    return false
+  }
+}
+
+function isEmail(value: string): boolean {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  return !isDID(value) && emailRegex.test(value)
+}
+
+export function ShareSpace({spaceDID}: {spaceDID: string}): JSX.Element {
   const [{ client }] = useW3()
   const [value, setValue] = useState('')
   const [downloadUrl, setDownloadUrl] = useState('')
+  const [sharedEmails, setSharedEmails] = useState<{ email: string, capabilities: string[] }[]>([])
+
+  useEffect(() => {
+    if (client && spaceDID) {
+      // Find all delegations where the spaceDID is present
+      const delegations = client.delegations()
+        .filter(d => d.capabilities.some(c => c.with === spaceDID))
+        .map(d => ({
+          email: DIDMailTo.toEmail(DIDMailTo.fromString(d.audience.did())),
+          capabilities: d.capabilities.map(c => c.can)
+        }))
+      setSharedEmails(delegations)
+    }
+  }, [client, spaceDID])
+
+  async function shareViaEmail(email: string): Promise<void> {
+    try {
+      if (!client) {
+        throw new Error(`Client not found`)
+      }
+
+      const space = client.spaces().find(s => s.did() === spaceDID)
+      if (!space) {
+        throw new Error(`Could not find space to share`)
```

Comment:
This should be enforced by `client.shareSpace(...)`

Reply:
Good point. I've already implemented that check in the client library, but I remember Hannah suggesting we delegate it to the caller side. I can reintroduce it as part of this refactor: [GitHub link](https://github.com/orgs/storacha/projects/1/views/1?pane=issue&itemId=80519054).

Reply:
I think it's redundant - `access.delegate(...)` will throw if the client cannot provide a proof.

Probably not a problem to leave here.